spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/metrics/MetricsUpdater.scala

```diff
@@ -0,0 +1,14 @@
+package com.datastax.spark.connector.metrics
+
+import org.apache.spark.SparkEnv
+
+trait MetricsUpdater {
+  def finish(): Long
+
+  def forceReport(): Unit = {
+    val env = SparkEnv.get
+    if (env != null) {
+      env.metricsSystem.report()
+    }
+  }
```

Comment:
I think you can simplify with:

     def forceReport(): Unit = Option(SparkEnv.get).map(_.env.metricsSystem.report)